src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.epics.js

```diff
@@ -12,75 +12,116 @@ import { actionCreator } from '../../../../actions/actions.utils';
 
 const trackerCaptureAppUrl = () => (process.env.REACT_APP_TRACKER_CAPTURE_APP_PATH || '..').replace(/\/$/, '');
 
-export const onScopeProgramFindUsingUniqueIdentifierEpic = (action$: InputObservable, store: ReduxStore) =>
+const filtersForUniqueIdSearchQuery = (searchTerm) => {
+    const fieldId = Object.keys(searchTerm)[0];
+    return [`${fieldId}:eq:${searchTerm[fieldId]}`];
+};
+
+const searchViaUniqueIdStream = (queryArgs, attributes, scopeSearchParam) =>
+    from(getTrackedEntityInstances(queryArgs, attributes)).pipe(
+        map(({ trackedEntityInstanceContainers }) => {
+            const searchResults = trackedEntityInstanceContainers;
+            if (searchResults.length > 0) {
+                const { id, tei: { orgUnit: orgUnitId } } = searchResults[0];
+                const oldTrackerCaptureAppUrl = trackerCaptureAppUrl();
+                const urlParameters = `/#/dashboard?tei=${id}&ou=${orgUnitId}&${scopeSearchParam}`;
+                window.location.href = `${oldTrackerCaptureAppUrl}${urlParameters}`;
+                return {};
+            }
+            return actionCreator(searchPageActionTypes.SEARCH_RESULTS_EMPTY)();
+        }),
+        startWith(actionCreator(searchPageActionTypes.SEARCH_RESULTS_LOADING)()),
+        catchError(() => of(actionCreator(searchPageActionTypes.SEARCH_RESULTS_ERROR)())),
```

Comment:
Already mentioned in the previous PR, but I would really like to keep the pattern with the actions file.

Reply:
yeah will change it :)